### PR TITLE
Add SameSite Restriction None

### PR DESCRIPTION
**Pull request by aanais**

### CODE REVIEW

**Comment by ocram** [comment]
Thanks!

This is important to add, of course, given the upcoming changes being introduced during Chrome 80+’s lifecycle (starting February) and other browsers following.

By the way, I’m not sure if we should return an error if `SameSite=None` is being set without `Secure` being set as well.

Introduction: https://blog.chromium.org/2019/10/developers-get-ready-for-new.html
RFC: https://tools.ietf.org/html/draft-west-cookie-incrementalism-00

**Comment by aanais** [comment]
Yes great advice I will check that and throw and exception if `SameSite=None` without `Secure `activation.
Thankt you !

File: src/Cookie.php [hunk 1]
```diff
@@ -362,6 +363,18 @@ public static function buildCookieHeader($name, $value = null, $expiryTime = 0,
...
+		elseif ($sameSiteRestriction === self::SAME_SITE_RESTRICTION_NONE) {
+			$headerStr .= '; SameSite=None';
+			//  According to Chrome 80+, secure should be set if SameSite is `None`
+			try {
+				if ($secureOnly != true) {
+					throw new \Exception('SecureOnly should be set on true if SameSite is set to `None`');
+				}
+				$headerStr .= '; SameSite=None';
+			} catch (\Exception $e) {
+				trigger_error($e->getMessage(), E_USER_ERROR );
```
Comment (aanais) [inline src/Cookie.php]: I don't know if I have to trigger a error or just a warning ?

Comment (aanais) [inline src/Cookie.php]: I don't know if I have to trigger an error or just a warning or a notice ?

**Comment by ocram** [comment]
Thanks!

Your original commit has been merged [here](https://github.com/delight-im/PHP-Cookie/commit/e912e89be9624e2a1aabce29a66cbd42be3b3e59).

Documentation and tests have been updated in:

 * https://github.com/delight-im/PHP-Cookie/commit/de980522a0d09da45d8118ad0e811321856a61d0
 * https://github.com/delight-im/PHP-Cookie/commit/1d31b13376ec18ed78229b77bbc16d81e0930d22
 * https://github.com/delight-im/PHP-Cookie/commit/10c9760629de0135ca4a01c4d264fb150af65499
 * https://github.com/delight-im/PHP-Cookie/commit/26612a0a1b8b9b51ee2ef6d972fa7910c7425fd7
 * https://github.com/delight-im/PHP-Cookie/commit/f5ea39079e74717db7f7a1cd0e68aa3e44d60814

Since the `None` value was introduced ([in October 2019](https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-03)) before the change of the defaults ([in November 2019](https://tools.ietf.org/html/draft-west-cookie-incrementalism-00)), we cannot throw an exception, trigger an error, or return `null` (which would be what the library does in other places).

So we’ll only be showing a warning: https://github.com/delight-im/PHP-Cookie/commit/1bf4c9b3d45433e0cd04b4cd114a66a75431885b

Documentation and tests updated in:

 * https://github.com/delight-im/PHP-Cookie/commit/c319bae581c095bf3b7a472ad52022ee0c92bc63
 * https://github.com/delight-im/PHP-Cookie/commit/a31fe0e13197a25ae8c389e51e31b1c575a796b6
 * https://github.com/delight-im/PHP-Cookie/commit/b6f869b69ac89d519ee1f38ea11671819cefbbf9

You should now be able to update to the latest version via Composer.

**Pull request closed without merging**